src/index.tsx

```diff
@@ -373,17 +393,30 @@ function QRCodeSVG(props: QRPropsSVG) {
       {image}
     </svg>
   );
-}
+});
 
 type RootProps =
   | (QRPropsSVG & {renderAs: 'svg'})
   | (QRPropsCanvas & {renderAs?: 'canvas'});
-const QRCode = (props: RootProps) => {
+const QRCode = React.forwardRef(function QRCode(
+  props: RootProps,
+  forwardedRef: React.ForwardedRef<HTMLCanvasElement | SVGSVGElement>
+) {
   const {renderAs, ...otherProps} = props;
   if (renderAs === 'svg') {
-    return <QRCodeSVG {...(otherProps as QRPropsSVG)} />;
+    return (
+      <QRCodeSVG
+        ref={forwardedRef as React.ForwardedRef<SVGSVGElement>}
+        {...(otherProps as QRPropsSVG)}
+      />
+    );
   }
-  return <QRCodeCanvas {...(otherProps as QRPropsCanvas)} />;
-};
+  return (
+    <QRCodeCanvas
+      ref={forwardedRef as React.ForwardedRef<HTMLCanvasElement>}
+      {...(otherProps as QRPropsCanvas)}
+    />
+  );
+});
```

Comment:
I'm sure this was messy to get to, but I'm going to strip out that default export so I'll end up removing this. To make your life easier, I'll merge your PR and then rebase my own.